### PR TITLE
UN-595: Adding analytics tags to record-links

### PR DESCRIPTION
**Pull request by nicklee**
Ticket URL: https://national-archives.atlassian.net/jira/software/c/projects/UN/boards/75?modal=detail&selectedIssue=UN-595

## About these changes

Adding data attrubute tags to the anchor link of record-links

## How to check these changes

Inspect the anchor link

## Before assigning to reviewer, please make sure you have

- [ ] Checked things thoroughly before handing over to reviewer.
- [ ] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [ ] Ensured that PR includes only commits relevant to the ticket.
- [ ] Waited for all CI jobs to pass before requesting a review.
- [ ] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: templates/records/blocks/record_links_block.html [hunk 1]
```diff
@@ -11,7 +11,9 @@
                     {% endif %}
                     <div class="record-links__item__text">
                         <p class="record-links__item__reference"><span class="sr-only">Link to record </span>FROM OUR COLLECTION: {{ item.record.reference_number }}</p>
-                        <p class="record-links__item__title"><a href="{{ item.record.url }}">{{ item.descriptive_title }}</a></p>
+                        <h2 class="record-links__item__title">
+                            <a data-component-name="{{ value.block.label }}" data-link-type="Link list" data-position="0" data-link="{{ item.descriptive_title }}" href="{{ item.record.url }}">{{ item.descriptive_title }}</a>
```
Comment (William-Blackie) [inline templates/records/blocks/record_links_block.html:15]: `{{ value.block.label }}` outputs: Record Links, which is correct in terms of your note of having two different SF elements, this being the newer one. It probably is fine but maybe flag this in the ticket too.

Also, I think that the requirement of `data-position=0` maybe a mistake, I think it should be `data-position={{ forloop.counter }}` as that would should the elements position in the list, what do you think?

**Review by William-Blackie — APPROVED** [review]
Hey @nicklee, nice work, I'll approve this as it does meet the brief but just double check as I think the requirements might not be correct. 

Thanks - Will.

File: templates/records/blocks/record_links_block.html (hunk 1 above)
Comment (nicklee) [reply templates/records/blocks/record_links_block.html:15]: 👍  I've queried the count on slack with Elise but haven't heard back, also had the same query on another metric ticket. I suspect you're right and it should be counter, so the analytics picks up which cards are clicked most often (1st, 2nd or 3rd) etc but will wait to hear back from Elise

Comment (nicklee) [reply templates/records/blocks/record_links_block.html:15]: I've just spoken to her and it needs to start at 0, so we need data-card-position="{{ forloop.counter0 }}"

**Review by jamesbiggs — APPROVED** [review]
Happy to approve this one as Will is off, but following the conversation and looking at the ticket it looks done to me!

**Pull request merged**